scripts/importer/data_importer/raw_data_importer.py

```diff
@@ -0,0 +1,502 @@
+        """
+        start = time.time()
+        header = [('chrom', str), ('pos', int), ('mean', float),
+                  ('median', float), ('cov1', float), ('cov5', float),
+                  ('cov10', float), ('cov15', float), ('cov20', float), ('cov25', float),
+                  ('cov30', float), ('cov50', float), ('cov100', float)]
+        logging.info("Inserting Coverage")
+        batch = []
+        last_progress = 0.0
+        counter = 0
+        with db.database.atomic():
+            for filename in self.settings.coverage_file:
+                for line in self._open(filename):
+                    line = bytes(line).decode('utf8').strip()
+                    if line.startswith("#"):
+                        continue
+
+                    data = {}
+                    for i, item in enumerate(line.strip().split("\t")):
+                        if i == 0:
+                            data['dataset_version'] = self.dataset_version
+                        data[header[i][0]] = header[i][1](item)
+
+                    if self.counter['coverage'] != None:
+                        counter += 1
+
+                    batch += [data]
+                    if len(batch) >= self.settings.batch_size:
+                        # re-format coverage for batch
+                        for i, item in enumerate(batch):
+                            batch[i]['coverage'] = [item['cov1'], item['cov5'], item['cov10'],
+                                                    item['cov15'], item['cov20'], item['cov25'],
+                                                    item['cov30'], item['cov50'], item['cov100']]
+                            del batch[i]['cov1']
+                            del batch[i]['cov5']
+                            del batch[i]['cov10']
+                            del batch[i]['cov15']
+                            del batch[i]['cov20']
+                            del batch[i]['cov25']
+                            del batch[i]['cov30']
+                            del batch[i]['cov50']
+                            del batch[i]['cov100']
+
+                        if not self.settings.dry_run:
+                            db.Coverage.insert_many(batch).execute()
+                        batch = []
+                        # Update progress
+                        if self.counter['coverage'] != None:
+                            progress = counter / self.counter['coverage']
+                            while progress > last_progress + 0.01:
+                                if not last_progress:
+                                    logging.info("Estimated time to completion: {}".format(self._time_to(start, progress)))
+                                    self._print_progress_bar()
+                                self._tick()
+                                last_progress += 0.01
+            if batch and not self.settings.dry_run:
+                db.Coverage.insert_many(batch)
```

Comment:
You don't do the batch processing of setting the `'coverage'` entry to be the array here that is done above on line 232-243.

Reply:
Moved the reformatting so its now performed before the data is added to batch.